### PR TITLE
Modernize cmake files

### PR DESCRIPTION
**Pull request by jbangelo**
Updated several aspects of the CMake files to conform with better practices. Also updated how dependencies are included to make it easier for upstream projects to specify their own versions when pulling in `libsettings` into their build.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,122 +1,42 @@
...
-# Some compiler options used globally
-if (NOT MSVC)
-  set(CMAKE_C_FLAGS
-      "-Wall -Wextra -Wno-strict-prototypes -Werror -fno-unwind-tables \
-       -fno-asynchronous-unwind-tables -Wimplicit -Wshadow -Wswitch-default \
-       -Wswitch-enum -Wundef -Wuninitialized -Wcast-align -Wformat=2 \
-       -Wimplicit-function-declaration -Wredundant-decls -Wformat-security \
-       -ggdb ${CMAKE_C_FLAGS}")
-endif (NOT MSVC)
+if (NOT LIBSBP_PATH)
```
Comment (jbangelo) [inline CMakeLists.txt]: I would have preferred to simply change to using `add_subdirectory()` and clearly and cleanly make `libsettings` depend on `libsbp` and `libswiftnav`. Unfortunately several of our projects try and define clang formatting targets with the same generic name (i.e. `clang-format-all`). This method is an ok middle ground for the time being, tough I think we should eventually re-examine our use of the clang format targets.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -17,9 +19,48 @@ add_library(settings
...
+    message(WARNING "No LibSbp include path found.")
+endif()
+
+if (EXISTS ${LIBSWIFTNAV_PATH}/include/)
+    target_include_directories(settings PRIVATE ${LIBSWIFTNAV_PATH}/include)
+else()
+    message(WARNING "No LibSwiftNav include path found.")
+endif()
+
+if (NOT MSVC)
```
Comment (pmiettinen) [inline src/CMakeLists.txt]: Maybe we can do something like
```
if (MSVC)
    target_compile_options(settings PRIVATE /Wall)
else ()
    target_compile_options(settings PRIVATE
            -Wall -Wextra -Wno-strict-prototypes -Werror -fno-unwind-tables
            -fno-asynchronous-unwind-tables -Wimplicit -Wshadow -Wswitch-default
            -Wswitch-enum -Wundef -Wuninitialized -Wcast-align -Wformat=2
            -Wimplicit-function-declaration -Wredundant-decls -Wformat-security
            -ggdb)
endif ()
```

`/WX` can't be added at this point as there are plenty of warnings in the msvc build, I have a separate issue for fixing those.

Comment (jbangelo) [reply src/CMakeLists.txt]: Sure thing!

**Comment by jbangelo** [comment]
Closing this PR as it's no longer necessary for my work.

**Pull request closed without merging**